hs_core/models.py

```diff
@@ -1515,10 +1705,73 @@ class ResourceFile(models.Model):
     fed_resource_file_name_or_path = models.CharField(max_length=255, null=True, blank=True)
     fed_resource_file_size = models.CharField(max_length=15, null=True, blank=True)
 
+    # we are using GenericForeignKey to allow resource file to be associated with any
+    # HydroShare defined LogicalFile types (e.g., GeoRasterFile, NetCdfFile etc)
+    logical_file_object_id = models.PositiveIntegerField(null=True, blank=True)
+    logical_file_content_type = models.ForeignKey(ContentType, null=True, blank=True,
+                                                  related_name="files")
+
+    logical_file_content_object = GenericForeignKey('logical_file_content_type',
+                                                    'logical_file_object_id')
+
+    mime_type = models.CharField(max_length=255, null=True, blank=True)
+
     @property
     def resource(self):
         return self.content_object
 
+    @property
+    def has_logical_file(self):
+        return self.logical_file is not None
+
+    @property
+    def logical_file(self):
+        return self.logical_file_content_object
+
+    @property
+    def logical_file_type_name(self):
+        return self.logical_file_content_object.__class__.__name__
+
+    @property
+    def has_generic_logical_file(self):
```

Comment:
What is the reason for the distinction between regular and "generic" logical files here?